### PR TITLE
detect-modbus/v1: add tests to modbus

### PR DESCRIPTION
**Pull request by TheKharleeci**
Tests included: 
DetectModbusTestAccess
DetectModbusTestFunction

### CODE REVIEW

File: tests/modbus/test.rules [hunk 1]
```diff
@@ -1,3 +1,5 @@
+alert modbus any any -> any any (msg:"Testing modbus code function"; modbus: access write; sid:6;)
```
Comment (jufajardini) [inline tests/modbus/test.rules:1]: This is maybe nit, but according to the documentation, this rule message should have a message like: `Testing modbus access` or something like that, I think...

cf https://suricata.readthedocs.io/en/latest/rules/modbus-keyword.html?highlight=modbus#modbus-keyword

File: tests/modbus/test.rules [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
+alert modbus any any -> any any (msg:"Testing modbus code function"; modbus: function 8; sid:7;)
```
Comment (jufajardini) [inline tests/modbus/test.rules:2]: Another nit: can we have the new signatures be put _after_ the ones that were already there and follow `sid` order? :)

**Review by jufajardini — COMMENTED** [review]
Looking good!

Two small notes in the comments.

I'm wondering if it wouldn't be good to use matches like we have for `sid` 4, with ` modbus.request.function_code:` as s-v tests also serve as use-cases for other folks learning about Suri. Let's see if the others think this would be a valid addition :)

File: tests/modbus/test.rules (hunk 2 above)
Comment (TheKharleeci) [reply tests/modbus/test.rules:2]: Ok I'll make that change

**Comment by TheKharleeci** [comment]
Followed By: #648

**Pull request closed without merging**